transport/server.cc

```diff
@@ -995,20 +997,38 @@ future<std::unique_ptr<cql_server::response>> cql_server::connection::process_st
 
 void cql_server::connection::update_scheduling_group() {
     if (_client_state.is_control_connection()) {
-        switch_tenant([this] (noncopyable_function<future<> ()> process_loop) -> future<> {
-            auto shg = _server._sl_controller.get_scheduling_group(qos::service_level_controller::driver_service_level_name);
-            _current_scheduling_group = shg;
-            co_return co_await _server._sl_controller.with_service_level(qos::service_level_controller::driver_service_level_name, std::move(process_loop));
-        });
+        update_service_lvl_scheduling_group();
+    } else if (_server._sl_controller.is_v2()) {
+        update_user_scheduling_group_v2(_client_state.user());
     } else {
-        switch_tenant([this] (noncopyable_function<future<> ()> process_loop) -> future<> {
-            auto shg = co_await _server._sl_controller.get_user_scheduling_group(_client_state.user());
-            _current_scheduling_group = shg;
-            co_return co_await _server._sl_controller.with_user_service_level(_client_state.user(), std::move(process_loop));
-        });
+        update_user_scheduling_group_v1(_client_state.user());
     }
 }
 
+void cql_server::connection::update_user_scheduling_group_v1(const std::optional<auth::authenticated_user>& usr) {
+    switch_tenant([this, usr] (this auto&& self, noncopyable_function<future<> ()> process_loop) -> future<> {
```

Comment:
`this auto&&`, I think this was supposed to be without `&&`. I'm not sure what was the point of using deducing this here (I'm not very familiar with this feature), but I know that we started using it to avoid the lambda coroutine fiasco (https://docs.seastar.io/master/tutorial.html#lambda-coroutines-c-23-and-later), and this requires passing `this` by value.

Reply:
correct this is a mistake, and capuring this is wrong now